docs/resources/firewalld.md.erb

```diff
@@ -4,34 +4,64 @@ title: About the firewalld Resource
 
 # firewalld
 
-Use the `firewalld` InSpec audit resource to test rules set to match IP address's with hostnames.
+Use the `firewalld` InSpec audit resource to test that firewalld is configured to allow and deny access to specific hosts, services and ports on a system.
 
-An firewalld rule specifies an IP address and what it's hostname is along with an aliase names it can have.
+A firewalld has a number of zones that can be configured to allow and deny access to specific hosts, services, and ports.
 
 ## Syntax
 
-Use the where clause to match a property to one or more rules in the hosts file.
-
     describe firewalld do
       it { should be_running }
-      its ('default_zone') { should eq 'public' }
+      its('default_zone') { should eq 'public' }
       it { should have_service_enabled_in_zone('ssh', 'public') }
-      it { should have_rule_enabled('rule family=ipv4 source address=192.168.0.14 accept', 'public') }
+      it { should have_rule_enabled('family=ipv4 source address=192.168.0.14 accept', 'public') }
+    end
+
+Use the where clause to test open interfaces, sources, and services in active zones.
+
+    describe firewalld.where { zone == 'public' } do
+      its('interfaces') { should cmp ['enp0s3', 'eno2'] }
+      its('sources') { should cmp ['192.168.1.0/24', '192.168.1.2'] }
+      its('services') { should cmp ['ssh', 'icmp'] }
     end
 
 ## Supported Properties
 
+### interfaces
+
+The `interfaces` property is used in conjunction with the where class to display open interfaces in an active zone.
+
+    describe firewalld.where { zone == 'public' } do
+      its('interfaces') { should cmp ['enp0s3', 'eno2'] }
+    end
+
+### sources
+
+The `sources` property is used in conjunction with the where class to display open sources in an active zone.
+
+    describe firewalld.where { zone == 'public' } do
+      its('souces') { should cmp ['192.168.1.0/24', '192.168.1.2'] }
+    end
+
+### services
+
+The `services` property is used in conjunction with the where class to display open services in an active zone.
+
+    describe firewalld.where { zone == 'public' } do
+      its('services') { should cmp ['ssh', 'icmp'] }
+    end
+
 ### default_zone
 
 The `default_zone` property displays the default active zone to be used.
 
-    its ('default_zone') { should eq 'public' }
+    its('default_zone') { should eq 'public' }
 
 ### default_zone
 
 The `active_zones` property displays the open zone and all its open interfaces
 
-    its ('active_zones') { should eq "public\n  interfaces: enp0s3\n" }
+    its('active_zones') { should eq "public\n  interfaces: enp0s3\n" }
```

Comment:
Is this a documentation error, or is this still expected behavior? I believe there was a prior comment from me that we shouldn't expect the user to format the output like this in order to check for active zones. It should only include the zone, no newlines or interfaces, etc.